DBM-Core/DBM-Core.lua

```diff
@@ -8584,9 +8587,27 @@ function DBM:GetBossHP(cIdOrGUID, onlyHighest)
 					bossHealth[cIdOrGUID] = hp
 				end
 				bossHealthuIdCache[cIdOrGUID] = unitId
+				bossNames[cIdOrGUID] = UnitName(unitId)
 				return hp, unitId, UnitName(unitId)
 			end
 		end
+		if not isRetail then
```

Comment:
Wouldn't it make sense to replace this entire classic block with just GetUnitIdFromCID or GetUnitIdFromGUID which both have automatic iteration selection for retail/classic, and will even use blizzards UnitTokenFromGUID function if available (guid checks only)

Based on your code i think you'd probably use GetUnitIdFromCID though which can be DBM or bossmodprototype/mod (I'll touch GetUnitIdFromCID up a little to not scan boss unit ids on classic era. although it doesn't error, it just wastes time)

Reply:
no, because i want this to happen as a fallback after scanning raid/party targets failed, not before